subprojects/plugin-development/src/test/groovy/org/gradle/plugin/devel/plugins/JavaGradlePluginPluginTest.groovy

```diff
@@ -224,7 +224,7 @@ class JavaGradlePluginPluginTest extends AbstractProjectBuilderSpec {
             _ * getConventionMapping() >> { Stub(ConventionMapping) }
         }
         project.tasks.remove(project.tasks.getByName(JavaGradlePluginPlugin.JAR_TASK))
-        project.tasks.add(mockJar)
+        project.tasks.addInternal(mockJar)
```

Comment:
this mockJar thing is kind of weird.  Do we need to do this in a unit test? It seems like `JavaGradlePluginPluginIntegrationTest` covers the tests that use mockJar well enough.

Reply:
Good point.  Removed.